test/gtest/core/test_alltoall.cc

```diff
@@ -35,35 +42,56 @@ class test_alltoall : public UccCollArgs, public ucc::test
             ctxs[i]->init_buf = ucc_malloc(
                 ucc_dt_size(dtype) * single_rank_count * nprocs, "init buf");
             EXPECT_NE(ctxs[i]->init_buf, nullptr);
-           for (int r = 0; r < nprocs; r++) {
+            for (int r = 0; r < nprocs; r++) {
                 size_t rank_size = ucc_dt_size(dtype) * single_rank_count;
                 alltoallx_init_buf(r, i,
-                                   (uint8_t*)ctxs[i]->init_buf + r * rank_size,
+                                   (uint8_t *)ctxs[i]->init_buf + r * rank_size,
                                    rank_size);
             }
-
-            UCC_CHECK(ucc_mc_alloc(
-                &ctxs[i]->dst_mc_header,
-                ucc_dt_size(dtype) * single_rank_count * nprocs, mem_type));
-            coll->dst.info.buffer = ctxs[i]->dst_mc_header->addr;
-            if (TEST_INPLACE == inplace) {
-                coll->mask  |= UCC_COLL_ARGS_FIELD_FLAGS;
-                coll->flags |= UCC_COLL_ARGS_FLAG_IN_PLACE;
-                UCC_CHECK(ucc_mc_memcpy(coll->dst.info.buffer, ctxs[i]->init_buf,
-                                        ucc_dt_size(dtype) * single_rank_count * nprocs,
-                                        mem_type, UCC_MEMORY_TYPE_HOST));
+            if (is_onesided) {
+                sbuf        = team->procs[i].p->onesided_buf[0];
+                rbuf        = team->procs[i].p->onesided_buf[1];
+                work_buf    = (long *)team->procs[i].p->onesided_buf[2];
+                work_buf[0] = -1; /* initialize for onesided a2a algorithm */
```

Comment:
@wfaderhold21 why this initialization to -1 is required? is it generic requirement for the user if onesided "work_buf" is provided or is it gtest specific?

Reply:
@vspetrov It's a generic requirement. The a2a algorithm is expecting the value to start at -1.

Reply:
@manjugv looks like this is then part of API/behavior which needs to be documented. Also i think we must have special constant for that (similar to shmem) instead of "-1" value.

Reply:
@wfaderhold21 "-1" might be an odd choice. Should we define a constant instead like we do in SHMEM?

Reply:
@vspetrov @manjugv I've added a line in the API stating that the buffer must be initialized to 0 and updated a2a code to match. Is this sufficient?

Reply:
As discussed offline, If "0" works for all the use-cases, this is the least intrusive way to do it, IMO. @vspetrov does this work for you?

Reply:
it does for me